psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/PowerFlowDataFactory.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.psse.model.pf.io;
+
+import com.powsybl.psse.model.PsseException;
+import com.powsybl.psse.model.PsseVersion;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public final class PowerFlowDataFactory {
+
+    private PowerFlowDataFactory() {
+    }
+
+    public static PowerFlowData create(String extension) {
+        // When only extension is given,
+        // We create PowerFlowData objects that are able to read the case identification
+        // To determine if the file is valid and get its version
+        if (extension.equalsIgnoreCase("rawx")) {
+            return new PowerFlowRawxDataAllVersions();
+        } else {
+            return new PowerFlowRawDataAllVersions();
+        }
+    }
+
+    public static PowerFlowData create(String extension, PsseVersion version) {
+        if (extension.equalsIgnoreCase("rawx")) {
+            return new PowerFlowRawxData35();
```

Comment:
I we can prepare the design for newer version?

Reply:
Not sure if this is what you were suggesting: I added a switch clause with explicit `V35` case. Rest of versions are unsupported at the moment.